src/main/java/org/opentripplanner/datastore/https/HttpsFileDataSource.java

```diff
@@ -89,7 +90,13 @@ public InputStream asInputStream() {
         throw new IllegalStateException(e.getLocalizedMessage(), e);
       }
     } else {
-      return in;
+      return ProgressTracker.track(
+        "Downloading %s".formatted(uri.toString()),
+        1000,
+        size(),
+        in,
+        m -> LOG.info(m)
+      );
```

Comment:
Currently progress tracking is done one abstraction level up. E.g. this will result in double tracking of OSM loading data. The benefit of adding this one level up, is that we have better/more detailed control of the context like use-case, expected speed and source. If added to the data-source it should be added as a generic feature that is opt-in and with configurable log message. Adding it to one DataSource and not the others is inconsistent. If I load it from HTTP I get tracking, but not if I load it from Google Cloud. Since we have few data-sources I think keeping this responsibility out of the DataSource component is a good thing. 

I suggests searching for the use of the `asInputStream()` and `asOutputStream()` and add progress tracking those locations instead - where the content is expected to take more than 5 seconds for a big source. In places where it is small, I would not add it - since it has a small overhead(synchronization - ignorable if we are fetching data outside mem).

Another reason for not adding it the the data-source is that the progress tracker always log, so for things that are so small that we do not want to track them, the tracker will create noice.

Another way to identify places to add tracking is to look through the logs and look for gaps in the logging that is more than 5 sec. If found, it is likely that there is a big task in the gap that shuld be tracked.